src/jaxsim/simulation/integrators.py

```diff
@@ -254,13 +195,29 @@ def body_fun(carry: Carry, xs: None) -> Tuple[Carry, None]:
         BW_vl_WB = (BW_Xv_W @ W_v_WB)[0:3]
 
         # Compute the derivative of the generalized position
-        d_pos_tf = jnp.hstack([BW_vl_WB, W_Qd_B, vel_tf[6:]])
+        d_pos_tf = (
+            jnp.hstack([BW_vl_WB, vel_tf[6:]])
+            if integrator_type is IntegratorType.EulerSemiImplicitManifold
+            else jnp.hstack([BW_vl_WB, W_Qd_B, vel_tf[6:]])
+        )
 
         # ------------------------------------
         # 3. Integrate the implicit velocities
         # ------------------------------------
 
         pos_tf = pos_t0 + sub_step_dt * d_pos_tf
+        sl = (
+            slice(3)
+            if integrator_type is IntegratorType.EulerSemiImplicitManifold
+            else slice(7)
+        )
```

Comment:
Probably is more readable removing the slice and extracting also $\mathbf{s}$ similarly as done with ${}^W \mathtt{Q}_B$.

```suggestion
        joint_positions = (
            pos_tf[3:7] ...
        )
```

Reply:
Addressed in https://github.com/ami-iit/jaxsim/pull/82/commits/e212c698518eedbe9ba39fff2965fb1618244572